com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -765,11 +765,17 @@ internal static void CheckOrphanChildren()
             }
         }
 
-        internal void InvokeBehaviourNetworkSpawn()
+        internal void InvokeBehaviourPreNetworkSpawn()
         {
             for (int i = 0; i < ChildNetworkBehaviours.Count; i++)
             {
                 ChildNetworkBehaviours[i].InternalOnNetworkSpawn();
+            }
+        }
+        internal void InvokeBehaviourPostNetworkSpawn()
```

Comment:
should we have tests for these? (I mean for both NetworkObject & SnapshotSystem)
like super-simple, bare-minimum tests.

Reply:
Yes. it's a bit more involved, though. What about a playmode test where an Object is spawned, with a NetVar and we simply check that the value is correctly set, at the moment it is spawned ? Maybe in OnNetworkSpawn ?

I could not come up with an edit mode test, or a mocked up test.

Reply:
I notice that some tests like `NetworkObjectOnSpawnTests` were previously failing, and are now passing. Does re-enabling them count as "tests for these" ?